scripts/repoint-frontend/repoint_frontend.sh

```diff
@@ -0,0 +1,203 @@
+}
+
+# $1: message, $2: info
+exit_on_failure() {
+  if [[ $? -ne 0 ]]; then
+    error "$1" "$2"
+    exit 1
+  fi
+}
+
+# check if in project root
+current_dir=$(basename "$PWD")
+if [[ "$current_dir" != "communications-manager-api" ]]; then
+  error "you must be in the project root directory to run this script"
+fi
+
+positional_arg_index=0
+
+# shifts positional args from arguments into named vars
+handle_positional_arg() {
+  case $((positional_arg_index++)) in
+    0) ticket_id="$1";;
+    1) environment="$1";;
+    *) error "too many positional arguments"
+  esac
+}
+
+# 0: remove mTLS (if set)
+# 1: create new branch for proxy modifications
+# 2: modify proxy files
+# 3: stage, commit, and push changes
+
+# sets list_steps_only to true, this is used by check_run_step
+list_steps() {
+  list_steps_only=$true
+  echo "STEPS:"
+}
+
+help() {
+  echo "This script automates the process of reconfiguring the 'communications-manager-api' APIs to point to a dynamic"
+  echo "backend environment in the 'comms-mgr' repository, rather than the default common backend in 'internal-dev'."
+  echo ""
+  echo "Usage: $0 [ticket ID] [environment] [options]"
+  echo ""
+  echo "Positional Arguments:"
+  echo "  ticket ID         Numeric only (e.g., '0000')"
+  echo "  environment       Environment identifier (e.g., 'de-gith1')"
+  echo ""
+  echo "Options:"
+  echo "  --help            Show this help message and exit."
+  echo "  --list-steps      List all steps and exit."
+  echo ""
+}
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --list-steps) list_steps;;
+        --help|-h) help; exit 0;;
+        -*) error "unknown option: $1" >&2;;
+        *) handle_positional_arg "$1";;
+    esac
+    shift || true
+done
+
+
+# Check AWS login status
+if ! aws sts get-caller-identity >/dev/null 2>&1; then
+  echo "You must have an active AWS SSO session to run this script."
+  exit 1
+fi
+
+# $1: step description
+check_run_step() {
+
+  # if only listing steps, list step and move on (do not run)
+  if [ $list_steps_only -eq $true ]; then
+    echo "  $((current_step++)): $1"
+    return $false
+  fi
+
+  echo ""
+  step "running: $1"
+  return $true
+}
+
+# doesn't validate the positional args if only listing steps
+if [[ $list_steps_only -eq $false ]]; then
+  # check if a ticket ID was provided
+  if [[ -z "$ticket_id" ]]; then
+    error "missing argument: ticket ID (numeric only)" "usage: $0 <ticket ID: 0000> <environment: de-gith1>"
+  fi
+
+  # check if a ticket environment was provided
+  if [[ -z "$environment" ]]; then
+    error "missing argument: environment" "usage: $0 <ticket ID: 0000> <environment: de-gith1>"
+  fi
+fi
+
+if [[ $list_steps_only -ne $true ]]; then
+  info "initial validation passed"
+fi
+
+if check_run_step "remove mTLS (if set)"; then
+  ./scripts/mtls/disable_mtls.sh $environment
+  exit_on_failure echo "Failed to disable mTLS, exiting..."
+fi
+
+if check_run_step "create new branch for proxy modifications"; then
```

Comment:
Can you remove these `check_run_step` conditions? I thought they were there previously for when you only wanted to run X step in the script

Reply:
Good point. I have removed the `check_run_step` function.